components/HeaderView.tsx

```diff
@@ -29,49 +29,101 @@ export function HeaderView() {
     },
   );
 
-  const onClickPollAdmin = () => {
-    router.push(`/polls/${slug}/admin`);
-  };
+  const onClickPollAdmin = () => router.push(`/polls/${slug}/admin`);
 
-  return (
-    <div className="self-start flex items-center justify-end w-full p-4 sticky top-0 bg-zinc-900 z-[60]">
-      <div className={clsx(!(isSignedIn && isAdmin) && "mr-auto")}>
-        <Link href="/" className="hover:opacity-50">
-          <Logo width={160} height={24} />
+  const toggleMenu = () => setIsMenuOpen(!isMenuOpen);
+
+  const closeMenu = () => setIsMenuOpen(false);
+
+  const renderNavLinks = (isMobile = false) => (
+    <>
+      <Link
+        href="/user/polls"
+        className={`${isMobile ? "block" : ""} text-white hover:opacity-50 text-sm`}
+        onClick={isMobile ? closeMenu : undefined}
+      >
+        My Polls
+      </Link>
+      <Link
+        href="/user/account"
+        className={`${isMobile ? "block" : ""} text-white hover:opacity-50 text-sm`}
+        onClick={isMobile ? closeMenu : undefined}
+      >
+        Account
+      </Link>
+      <Button
+        variant="pill"
+        size="pill"
+        className={isMobile ? "w-full" : "mr-2"}
+        asChild
+      >
+        <Link
+          href="/new-poll"
+          prefetch={false}
+          onClick={isMobile ? closeMenu : undefined}
+        >
+          <PlusCircle className="w-3 mr-2" /> Create Poll
         </Link>
-      </div>
+      </Button>
+    </>
+  );
```

Comment:
A function which returns react elements is a react component, there's no reason to create it on every render. Move both of these into separate components:
```tsx
function NavLinks({ isMobile, closeMenu }) { ... }
function MobileMenu({ toggleMenu, isMenuOpen }) { ... }
```